Automattic/jetpack
# REST API: Add view_hosting capability to sites endpoints

#### Changes proposed in this Pull Request:
Adds a new `view_hosting` capability to the response of all the `/sites/:site` endpoints that indicates if the user can access the hosting section in Calypso. It defaults to false since Jetpack sites are not eligible for that feature, but it can be extended (it'll be needed in https://github.com/Automattic/wp-calypso/issues/38484). D37177-code is the counterpart WP.com diff of this change.

#### Testing instructions:
* Apply D37177-code and sandbox the API.
* Make a `GET` request to `https://public-api.wordpress.com/rest/v1.2/sites/:site` for a Jetpack site.
* Check that the `capabilities` field includes a new `view_hosting` key with a `false` value.

#### Proposed changelog entry for your changes:
* No changelog needed
